src/python_testing/mdns_discovery/utils/net_utils.py

```diff
@@ -0,0 +1,75 @@
+#
+#    Copyright (c) 2025 Project CHIP Authors
+#    All rights reserved.
+#
+#    Licensed under the Apache License, Version 2.0 (the "License");
+#    you may not use this file except in compliance with the License.
+#    You may obtain a copy of the License at
+#
+#        http://www.apache.org/licenses/LICENSE-2.0
+#
+#    Unless required by applicable law or agreed to in writing, software
+#    distributed under the License is distributed on an "AS IS" BASIS,
+#    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#    See the License for the specific language governing permissions and
+#    limitations under the License.
+#
+
+import ipaddress
+import logging
+
+import ifaddr
+from zeroconf import InterfaceChoice
+
+logger = logging.getLogger(__name__)
+
+
+def get_host_ipv6_addresses():
+    """
+    Retrieves a list of valid IPv6 addresses for all network interfaces on the host.
+
+    This function scans all network adapters for assigned IPv6 addresses and filters out 
+    unusable ones such as loopback (::1) and invalid addresses. It includes both link-local 
+    (with interface scope) and routable (ULA or global) IPv6 addresses.
+
+    Returns:
+        list[str] | InterfaceChoice:
+            - A list of formatted IPv6 addresses (with scope if required), or
+            - InterfaceChoice.All if no usable IPv6 addresses are found, in which case Zeroconf's 
+              default interface selection behavior will apply.
+    """
+    results = []
+
+    for adapter in ifaddr.get_adapters():
+        # Get list of IPv6 addresses for the interface
+        for ip in adapter.ips:
+            # Get IPv6 address and interface
+            if isinstance(ip.ip, tuple) and ':' in ip.ip[0]:
```

Comment:
why is there a `:` check here? IPv6 would have the ip.ip as a tuple according to 
https://pythonhosted.org/ifaddr/#ifaddr.IP

the : should be explained here ... I am unclear what it means as every ipv6 address should have it. Would a cast to IPv6Address not be sufficient?